### ISSUE TITLE
Custom node handles in a sub-flow not responding to drag correctly

### ISSUE THREAD

**Issue by HazelDaniel** [xyflow/xyflow#4746]
### What platform were you using when you found the bug?

- React Flow / Svelte Flow version: ^12.3.2
- Browser and version: Chrome 125.0.6422.60 (Official Build) (64-bit)
- OS and version: Linux OS, 6.6.9-amd64

### Describe the Bug

i defined two custom nodes in my application; a `groupTableNode` type to group the sub-flows and N `tableNode` children serving as the sub-flows.

i implemented the handles for these nodes as demonstrated on the docs but they do not work as intended.
here's a snippet of a handle:
![handle-drag-component-code](https://github.com/user-attachments/assets/48872a44-faed-4a6b-afc0-93c7004a5a67)

defined for a `tableNode` node:
![node-full-code](https://github.com/user-attachments/assets/a915f22c-f73e-470c-b90b-5d4cd8fa2f37)

### Steps to reproduce the bug or issue

N/A

### Expected behavior

i expect each handle as shown in the diagram below to be able to produce edges when dragged but no matter which handle i drag (on the nodes in the same sub-flow), only the handles of the first node in the sub-flow respond to the drag of other handles (other handles as in handles in the subflow having the same position attribute).

### Screenshots or Videos

case 1 (dragging the handles of the first node in the sub-flow):

![handle-drag-example](https://github.com/user-attachments/assets/ccc486be-c4e9-4961-a72e-69c658b59a4d)

case 2 (dragging the handles of other nodes in the sub-flow):
![handle-drag-issue-result](https://github.com/user-attachments/assets/c2d9b8f7-d48d-4e4f-9cf2-5f3bd4cfd25d)

**Comment by peterkogo** [comment]
Is it possible you are offsetting the tableNodes with some styles?
The way React Flow works is that it renders the nodes and then measures the position of the handles relative to the node. Because this is a computationally intensive operation they are only remeasured when the node resizes. 

Can you try using the [useUpdateNodeInternals](https://reactflow.dev/api-reference/hooks/use-update-node-internals) hook to see if the issue persists?

**Comment by HazelDaniel** [comment]
yes, i'm offsetting the position of the `tableNode`s  (vertically) with styles since the sub-flow group is only a logical group and as such, there's no way for the DOM to render the layout flow as intended - unless of course, there's a cleaner workaround i don't know about.

but this is how the positioning logic looks as of now:
![nodes-state](https://github.com/user-attachments/assets/aee58160-fabe-4ade-a227-472f8a2efd06)


And based on your suggestion, I have tried using the useUpdateInternals hook but to no avail.
![use-update-internal-nodes-adjustment](https://github.com/user-attachments/assets/f98bdcd0-6c81-4794-9b59-83dbbd89a123)

**Comment by peterkogo** [comment]
> yes, i'm offsetting the position of the tableNodes (vertically) with styles since the sub-flow group is only a logical group and as such, there's no way for the DOM to render the layout flow as intended

This is the issue. React Flow expects the nodes to be at the x,y position you set in the nodes array not some arbitrary offset you apply through styling. You have to calculate the offset in JavaScript and apply it to the x,y positions of the nodes.

I will close this issue as this is not a bug report. However, you can head over to our [very active Discord](https://discord.gg/RVmnytFmGW) where it's easier to get help :)

**Comment by HazelDaniel** [comment]
Alright, thanks for the insight.
